Configuration/PyReleaseValidation/python/MatrixInjector.py

```diff
@@ -408,6 +438,9 @@ def prepare(self, mReader, directories, mode='init'):
                                     if setPrimaryDs:
                                         chainDict['nowmTasklist'][-1]['PrimaryDataset']=setPrimaryDs
                                 nextHasDSInput=None
+                                if 'GPU' in step and self.RequiresGPU == 'required':
+                                    chainDict['nowmTasklist'][-1]['RequiresGPU'] = self.RequiresGPU
+                                    chainDict['nowmTasklist'][-1]['GPUParams']=json.dumps(self.defaultGPUParams)
```

Comment:
I don't understand this part.
Why not simply
```suggestion
                                if self.RequiresGPU != 'forbidden':
                                    chainDict['nowmTasklist'][-1]['RequiresGPU'] = self.RequiresGPU
                                    chainDict['nowmTasklist'][-1]['GPUParams']=json.dumps(self.defaultGPUParams)
```
(and similarly below) ?

Reply:
Also, why so nested in all the checks, instead of simply being done for all steps ?

Reply:
I code in this way following the discussion in https://github.com/dmwm/WMCore/issues/10393#issuecomment-820159056, to be flexible in the task/step level. In case of taskchain, one can do GEN-SIM in non-GPU env, while HLT in GPU env, for example.

Reply:
IMHO that is not going to be maintainable - we cannot add "GPU" in the name of all the steps that we want to (potentially) run on a GPU-equipped node.

For example, soon enough the HLT step of any Run-3 workflow will be able to run on GPUs; so it could make sense to submit jobs with `--gpu optional`, but I doublt we want to rename everything adding GPU in its name.

Reply:
> Here's the proposed update to the default values and help messages.
> 
> I didn't check if it runs !

Thanks very much @fwyzard 
I've implemented all suggestions, but leave checking on `GPU` on step name for now. We can open a discussion with @amaltaro